frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx

```diff
@@ -0,0 +1,342 @@
+  DATE_ACCESSORS,
+  POSITIONAL_ACCESSORS,
+} from "../../constants/accessors";
+import { getWaterfallColors } from "../../lib/colors";
+import Values from "../Values";
+import { createXScale } from "../XYChart/utils";
+
+const layout = {
+  width: 540,
+  height: 300,
+  margin: {
+    // Add some margin so when the chart scale down,
+    // elements that are rendered at the top of the chart doesn't get cut off
+    top: 12,
+    left: 55,
+    right: 40,
+  },
+  barPadding: 0.2,
+  labelPadding: 12,
+  strokeDasharray: "4",
+  numTicks: 4,
+  maxTickWidth: 100,
+};
+
+// If inner chart area is smaller than this then it will look cramped
+const MIN_INNER_HEIGHT = 250;
+const MAX_EXTRA_HEIGHT = 40;
+// The default value for tick length
+const TICK_LENGTH = 8;
+const VALUES_MARGIN = 6;
+
+// Since we're using JSDoc instead, which provide more static type checking than PropTypes.
+/* eslint-disable react/prop-types */
+/**
+ *
+ * @param {import("./types").WaterfallChartProps} props
+ * @returns {JSX.Element}
+ */
+function WaterfallChart({
+  data,
+  type,
+  accessors = getDefaultAccessors(type),
+  settings,
+  labels,
+  getColor,
+}) {
+  const chartStyle = {
+    fontFamily: "Lato, sans-serif",
+    axes: {
+      color: getColor("text-light"),
+      ticks: {
+        color: getColor("text-medium"),
+        fontSize: 12,
+      },
+      labels: {
+        color: getColor("text-medium"),
+        fontSize: 14,
+        fontWeight: 700,
+      },
+    },
+    value: {
+      color: getColor("text-dark"),
+      fontSize: 12,
+      fontWeight: 800,
+      stroke: getColor("white"),
+      strokeWidth: 3,
+    },
+  };
+
+  const axesProps = {
+    stroke: chartStyle.axes.color,
+    tickStroke: chartStyle.axes.color,
+  };
+
+  const valueProps = {
+    fontSize: chartStyle.value?.fontSize,
+    fontFamily: chartStyle.fontFamily,
+    fontWeight: chartStyle.value?.fontWeight,
+    letterSpacing: 0.5,
+    fill: chartStyle.value?.color,
+    stroke: chartStyle.value?.stroke,
+    strokeWidth: chartStyle.value?.strokeWidth,
+  };
+
+  if (type === "timeseries") {
+    data = sortTimeSeries(data);
+  }
+  const entries = calculateWaterfallEntries(
+    data,
+    accessors,
+    settings?.showTotal,
+  );
+
+  const isVertical = type === "timeseries" ? false : entries.length > 10;
+  const xTickWidth =
+    type === "timeseries"
+      ? // We don't know the width of the time-series label because it's formatted inside `<AxisBottom />`.
+        // We could extract those logic out, but it's gonna be nasty, and we didn't need `xTickWidth` for time-series anyway.
+        0
+      : getXTickWidth(
+          data,
+          accessors,
+          layout.maxTickWidth,
+          chartStyle.axes.ticks.fontSize,
+        );
+
+  const getXTickProps =
+    type === "timeseries"
+      ? ({ formattedValue, ...props }) => {
+          return {
+            ...props,
+            children: formatTimescaleWaterfallTick(formattedValue, settings),
```

Comment:
This is extracted from https://github.com/metabase/metabase/blob/29812d246fcad8359ec1a362a74b8d9a708e9868/frontend/src/metabase/static-viz/components/TimeSeriesWaterfallChart/TimeSeriesWaterfallChart.jsx#L145 and make it looks similar to the `getXTickProps` used defined in `CategoricalBarChart`, so they're compatible.